src/redux/profile/profileMiddleware.ts

```diff
@@ -0,0 +1,42 @@
+import { createAsyncThunk } from '@reduxjs/toolkit';
+
+import { api } from '@/api/api';
+import { onOpenConnection } from '@/redux/connection/connectionSlice';
+import { AppStartListening } from '@/redux/listenerMiddleware';
+
+import { setMenuItemVisible } from '../local/localSlice';
+
+export interface ProfileState {
```

Comment:
I just realized that there is only a middleware for this. Why is that? 

It would match the the other redux code more if this data structure was located in a file `profileSlice.ts`

Reply:
We don't need to store the profile data because the data is actually handled by other slices so this middleware is really only for getting the topic. It is possible that we might want to add things in the future that we want to store but I think we can add that when needed